core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -554,6 +555,32 @@ public String authority() {
     }
   }
 
+  private final class DirectFallbackExecutor implements Executor {
+    @Override
+    public void execute(final Runnable command) {
+      final boolean[] accepted = new boolean[1];
+      try {
+        ManagedChannelImpl.this.executor.execute(new Runnable() {
+          @Override
+          public void run() {
+            accepted[0] = true;
+            command.run();
+          }
+        });
+      } catch (RejectedExecutionException e) {
```

Comment:
I agree that none of these solutions actually fix the race. But with the rejection check, the only unresolved race is very very benign as follows:
The `executor.execute()` is called and throws a `RejectedExecutionException` for whatever unlikely reason before the channel is terminated, and channel is terminated when we check it, so we don't throw this exception, and call `command.run()` directly.

Other cases are handled gracefully.